executor_test.go

```diff
@@ -33,6 +35,20 @@ import (
 	"github.com/pkg/errors"
 )
 
+var (
+	TempDir *string
+)
+
+func init() {
+	tdflag := flag.Lookup("temp-dir")
+	if tdflag == nil {
+		TempDir = flag.String("temp-dir", "", "Directory in which to place temporary data (e.g. for benchmarking). Useful if you are trying to benchmark different storage configurations.")
+	} else {
+		s := tdflag.Value.String()
```

Comment:
so, this whole init() is fun... `TempDir` (and its flag) are actually declared in two files. Both are _test files, so it's only around for tests, but one is in package pilosa, and the other is in pilosa_test. Declaring the same flag twice leads to a panic, so in the pilosa_test package, we guard by looking up the flag first to see if it's already defined, and then only define it if it isn't.

Reply:
Could we avoid the init() call altogether by making funcs like `pilosa.GetTempDir()` and `pilosa.GetTempFile()`?

Reply:
I got rid of the init. I think the way I've done this is slightly better than defining a func to call at each invocation of `ioutil.Temp*`, because we need something that defines the flag at startup time. feel free to disagree though

Reply:
I didn't actually realize you can call functions from the top level like that. Since that is essentially the same as init() in terms of startup side-effects, it's interesting that there is no linter that checks for that.